src/pages/Login/Login.tsx

```diff
@@ -0,0 +1,106 @@
+import "./Login.scss";
+import Button from "../../components/Button/Button";
+import arrowLeft from "../../assets/images/arrow-left.png";
+import { ChangeEvent, FormEvent, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { signInWithEmailAndPassword } from "firebase/auth";
+import { auth } from "../../firebase";
+import { AuthError } from "firebase/auth";
+
+const emptyFormData = {
+  email: "",
+  password: "",
+};
+
+type LoginProps = {
+  setUserUID: (arg0: string) => void;
+};
+
+export const Login = ({ setUserUID }: LoginProps) => {
+  const navigate = useNavigate();
+  const [formData, setFormData] = useState(emptyFormData);
+  const [formErrorMessage, setFormErrorMessage] = useState("");
+  const [accessToken, setAccessToken] = useState<string>("");
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  // Setting accessToken allows for easier login.
+  useEffect(() => {
+    localStorage.setItem("accessToken", JSON.stringify([accessToken]));
+  }, [accessToken]);
+
+  const onLogin = async (e: FormEvent) => {
+    e.preventDefault();
+    try {
+      const userCredential = await signInWithEmailAndPassword(
+        auth,
+        formData.email,
+        formData.password
+      );
+      console.log(userCredential);
+      setUserUID(userCredential.user.uid);
+      const userIDToken = await userCredential.user.getIdToken();
+      setAccessToken(userIDToken);
```

Comment:
I would replace this with what you have in the useEffect.
`localStorage.setItem("accessToken", JSON.stringify([accessToken]));`

You navigate away from the page so I think using state and a useEffect adds complexity.